pymeasure/instruments/srs/ldc500series.py

```diff
@@ -0,0 +1,516 @@
+        cast=int,
+    )
+
+    def check_errors(self):
+        """Read all errors from the instrument.
+
+        :return: List of [``last_execution_error``, ``last_command_error``]
+        """
+        return [self.last_execution_error, self.last_command_error]
+
+    def check_set_errors(self):
+        """Check for errors after having set a property, and raise an error if any are present."""
+        # FIXME: The LDC500 has a delay between a command being executed and an error being logged.
+        #        *OPC? didn't work to wait for the previous command, so instead as short a delay
+        #        as possible has been put in. It's not a nice solution and any ideas would be
+        #        appreciated.
+        sleep(0.015)
+        errors = self.check_errors()
+        if errors == [0, 0]:
+            return []
+        raise Error(f"Error setting value: {errors}")
+
+
+class LDC500Series(LDC500SeriesBase):
+    """Represents an SRS LDC500Series laser diode controller
+    and provides a high-level interface for interacting with the instrument.
+
+    Attributes
+    ----------
+    ld : LDC500SeriesLDSubsystem
+        Provides access to laser diode control (e.g. ``ldc.ld.mode``)
+    pd : LDC500SeriesPDSubsystem
+        Provides access to photodiode control (e.g. ``ldc.pd.power``)
+    tec : LDC500SeriesTECSubsystem
+        Provides access to TEC control (e.g. ``ldc.tec.current``)
+
+    Example
+    ~~~~~~~
+
+    .. code-block:: python
+
+        ldc = LDC500Series("GPIB::5")
+        ldc.ld.enabled = True
+        temperature = ldc.tec.temperature
+
+    Glossary:
+    ~~~~~~~~~
+
+        - CC: Constant Current Mode
+        - CP: Constant Power Mode
+        - CT: Constant Temperature Mode
+        - LD: Laser diode
+        - PD: Photodiode
+        - TEC: Thermo-electric controller
+    """
+
+    def __init__(self, adapter, name="LDC500Series", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        self.ld = LDC500SeriesLDSubsystem(self)
+        self.pd = LDC500SeriesPDSubsystem(self)
+        self.tec = LDC500SeriesTECSubsystem(self)
+
+
+class LDC500SeriesLDSubsystem(LDC500SeriesBase):
+    """Subsystem of LDC500Series for control of the laser-diode (LD)."""
+
+    def __init__(self, adapter, name="LDC500Series LD Subsystem", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    interlock_closed = Instrument.measurement(
+        "ILOC?",
+        """Get the status of the interlock, True if closed, False if open.
+        Laser will only operate with a closed interlock.""",
+        values={True: "CLOSED", False: "OPEN"},
+        map_values=True,
+    )
+
+    enabled = Instrument.control(
+        "LDON?",
+        "LDON %s",
+        """Control whether the laser diode current source is enabled (bool).""",
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    mode = Instrument.control(
```

Comment:
as an idea: if you call it `CC_mode_enabled` (or similar), you can use a boolean to control it